quarkchain/cluster/shard_state.py

```diff
@@ -344,6 +348,14 @@ def __create_evm_state(
         state.sender_disallow_map = sender_disallow_map
         if timestamp:
             state.timestamp = timestamp
+        # iterate until reaches genesis or header list reaches 256
+        # since most headers are in LRU cache, this should not affect performance too much
+        while block_hash and len(state.prev_headers) < 256:
+            h = self.db.get_minor_block_header_by_hash(block_hash)
+            if not h:
+                break
+            state.prev_headers.append(h)
```

Comment:
Why not use state.add_block_header?  Looks like add_block_header appendleft, but this appendright, but I cannot see why we need to appendleft?  Perhaps we could replace appendright to appendleft in add_block_header if appendleft is not used anywhere?

Reply:
note the order of `prev_headers` is descending, i.e. header at height 100, heigh 99, etc. both this part and `evm_state.add_block_header(block.header)` will keep this invariant (`appendleft` is used in `evm_state.add_block_header` since the passed block is usually at the highest height).